hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+                   the SPI_HOST IP, rather they are multiplexed out to match the
+                   on-chip inputs `pt_sck_i`, `pt_cs_ni`, `pt_sdt_i`, `pt_sdo_i`,
+                   and `pt_sdi_o`.  (Since the `sd` bus is bidirectional, there are
+                   separate input, output and tri-state controls for this bus.)''',
+          resval: "0x0"
+        },
+        { bits: "24:16",
+          name: "TX_WATERMARK"
+          desc: '''The IP can optionally send an interrupt if the
+                   depth of the TX FIFO is ever less than
+                   TX_WATERMARK words (32b each). Set to zero
+                   to temporarily disable interrupts.'''
+          resval: "0"
+        },
+        { bits: "15:9",
+          name: "RX_WATERMARK"
+          desc: '''The IP can optionally send an interrupt if the
+                   depth of the RX FIFO is ever greater than
+                   RX_WATERMARK words (32b each).'''
+          resval: "127"
+        },
+        { bits: "8",
+          name: "MANCS_EN",
+          desc: '''Enable Manual Chip Select. Typically each command
+                   manages the CS lines automatically, issuing a command
+                   to exactly one device, and lowering and raising the
+                   appropriate CS line after transmitting or receiving the
+                   necessary bytes.  Asserting this bit overrides this
+                   automatic selection, instead setting the CS lines to
+                   reflect the contents of the MANUAL_CS field.'''
+          resval: "0x0"
+        },
+        { bits: "7:0",
+          name: "MANUAL_CS"
+          desc: '''When MANCS_EN = 1, the cs_n lines are directly to the
+                   value of MANUAL_CS.'''
+          resval: 0
+        }
+      ]
+    },
+    { name: "STATUS",
+      desc: "Status register",
+      swaccess: "ro",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "31",
+          name: "READY",
+          desc: '''When high, indicates the SPI host is ready to receive
+                   commands. Writing to SPICMD when RDY is low is an error,
+                   and will trigger an interrupt.
+                   ''',
+          resval: "0x0"
+        },
+        { bits: "30",
+          name: "ACTIVE",
+          desc: '''When high, indicates the SPI host is processing a previously
+                   issued command.'''
+          resval: "0x0"
+        },
+        { bits: "29",
+          name: "TXFULL",
+          desc: '''When high, indicates that the transmit data fifo is full.
+                   Any further writes to TXD will create an error interrupt.
+                '''
+          resval: "0x0"
+        },
+        { bits: "28",
+          name: "TXEMPTY",
+          desc: '''When high, indicates that the transmit data fifo is empty.
+                   And pending commands may have to wait for more data.
+                '''
+          resval: "0x0"
+        },
+        { bits: "27"
```

Comment:
so what happens if there is no data? does the transaction terminate immediately? Or do you have some means of sending continued dummy data?

The latter has actually been useful to software in the past.

Reply:
It currently stalls.  There is no provision currently to send dummy data.

Reply:
@garrett, @a-pronin: Opinions?  I personally find the idea of a SPI that sends garbage data when empty quite alarming.

Reply:
@gkelly, I mean

Reply:
Actually there is an option to send explicitly send dummy cycles.  If you want to set real data and then dummy cycles you can use the CSAAT mechanism to piece together something unique like this.

Reply:
yeah let's ping them and see what they think.  I should have said that it's not "garbage" data.. more like "default" data. 
I know this can be useful in construction of certain protocols, but the software team will know best.

Reply:
btw, also fine IMO to break this into a separate issue and not hold the PR.

Reply:
Sounds good.  Made a new issue: https://github.com/lowRISC/opentitan/issues/5133